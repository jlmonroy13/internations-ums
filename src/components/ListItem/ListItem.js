import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';

function ListItem({
  data,
  type,
  page,
  urlId,
  deleteGroupFromUser,
  deleteUserFromGroup,
  itemsLength,
}) {
  const { urlPhoto, name, description, email, firstName, lastName, id } = data;
  const fullName = `${firstName} ${lastName}`;
  let link = type;
  if (page === 'details' && type === 'user') link = 'group';
  if (page === 'details' && type === 'group') link = 'user';

  function testingClick() {
    if (type === 'user') {
      deleteGroupFromUser(urlId, id);
    } else {
      deleteUserFromGroup(urlId, id);
    }
  }
  return (
    <li className="list__item">
      <Link className="list__item-link" to={`/${link}/${id}`} />
      <img src={urlPhoto} alt="Group photo" className="list__item-image" />
      <div className="list__item-info">
        <span className="list__item-text">{name || fullName}</span>
        <span className="list__item-text list__item-text--small list__item-text--dots">
          {description || email}
        </span>
        {page === 'details' && ((type === 'user' && itemsLength !== 1) || type === 'group') ? (
          <span className="list__item-btn icon icon--exit" onClick={testingClick} />
        ) : null}
      </div>
    </li>
  );
}

ListItem.propTypes = {
  data: PropTypes.object,
  type: PropTypes.string.isRequired,
  page: PropTypes.string.isRequired,
  urlId: PropTypes.string,
  itemsLength: PropTypes.number,
  deleteGroupFromUser: PropTypes.func,
  deleteUserFromGroup: PropTypes.func,
};

export default ListItem;
