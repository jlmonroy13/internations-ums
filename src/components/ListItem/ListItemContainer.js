import { connect } from 'react-redux';
import { deleteUserFromGroup } from '../../actions/group';
import { deleteGroupFromUser } from '../../actions/user';
import ListItem from './ListItem';

function mapDispatchToProps(dispatch) {
  return {
    deleteGroupFromUser: (contactId, groupId) => dispatch(deleteGroupFromUser(contactId, groupId)),
    deleteUserFromGroup: (groupId, contactId) => dispatch(deleteUserFromGroup(groupId, contactId)),
  };
}

export default connect(null, mapDispatchToProps)(ListItem);
