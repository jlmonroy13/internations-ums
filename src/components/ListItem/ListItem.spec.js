import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from './ListItem';

describe('<ListItem />', () => {
  it('should render the list item component', () => {
    const data = {
      urlPhoto: '',
      name: '',
      description: '',
      email: '',
      firstName: '',
      lastName: '',
      id: '',
    }
    const wrapper = renderer.create(<ListItem page="" type="" data={data}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
