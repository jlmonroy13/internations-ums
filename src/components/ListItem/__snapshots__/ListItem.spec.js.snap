// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should render the list item component 1`] = `
<li
  className="list__item"
>
  <a
    className="list__item-link"
    onClick={[Function]}
    style={Object {}}
  />
  <img
    alt="Group photo"
    className="list__item-image"
    src=""
  />
  <div
    className="list__item-info"
  >
    <span
      className="list__item-text"
    >
       
    </span>
    <span
      className="list__item-text list__item-text--small list__item-text--dots"
    >
      
    </span>
  </div>
</li>
`;
